### ISSUE TITLE
Bug: `Boxes.__getitem__` drops the list padding metadata, so a sliced list-backed object exports its zero pad rows as boxes

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4179]
## Summary

`Boxes.__getitem__` (`kornia/geometry/boxes.py:237` at `e0969438`) rebuilds the result from the raw padded data and copies only `_mode`:

```python
def __getitem__(self, key):
    new_box = type(self)(self._data[key], False)
    new_box._mode = self._mode
    return new_box
```

`_N`, the per-element padding counts recorded when the object was built from a list, is not carried over, so the sliced object no longer knows it is list-backed. `to_tensor` then returns the **padded tensor** instead of the trimmed list, and the zero pad rows come back as boxes. `clone`, `merge` and `index_put` all preserve `_N`; only indexing loses it. Nothing raises.

## Reproduction

```python
import torch
from kornia.geometry.boxes import Boxes

first = torch.tensor([[[1.0, 2.0], [4.0, 2.0], [4.0, 3.0], [1.0, 3.0]]])   # 1 box
lb = Boxes([first, torch.cat([first, first])])                              # _N == [1, 0], data (2, 2, 4, 2)

print(lb._N, [t.tolist() for t in lb.to_tensor("xyxy")])
sliced = lb[0:1]
print(sliced._N, type(sliced.to_tensor("xyxy")).__name__, sliced.to_tensor("xyxy").tolist())
print([t.tolist() for t in sliced.get_boxes_shape()])
print(lb.clone()._N, lb.index_put((torch.tensor([0]), torch.tensor([0])), torch.zeros(4, 2))._N)
```

## Observed

```
[1, 0] [[[1.0, 2.0, 5.0, 4.0]], [[1.0, 2.0, 5.0, 4.0], [1.0, 2.0, 5.0, 4.0]]]
None Tensor [[[1.0, 2.0, 5.0, 4.0], [0.0, 0.0, 1.0, 1.0]]]
[[[2.0, 1.0]], [[4.0, 1.0]]]
[1, 0] [1, 0]
```

The unsliced object exports a list with row 0 trimmed to its one real box. `lb[0:1]` exports a `(1, 2, 4)` tensor in which the phantom padding box `[0, 0, 1, 1]` sits next to the real one, and `get_boxes_shape` reports it as a 1-by-1 box. `lb[1:2]` happens to look right only because that row's pad count was 0, so whether a slice is corrupt depends on how ragged the original list was.

## Expected

`__getitem__` carries the padding metadata through: `new_box._N = self._N[key]` for a slice or index tensor on the batch axis (a plain `int` key drops the batch axis and returns an unbatched object, which has no `_N`), or, if indexing a list-backed object is not meant to be supported, it raises rather than returning a silently padded tensor. The batched export should be the same list as `[lb.to_tensor("xyxy")[i] for i in range(...)]` restricted to the selected rows.

## Context

Found in round 9 of #4060 (bbox/boxes convention blocks, batch 4a), which documents the trimmed-list contract of `to_tensor(as_padded_sequence=False)` and `get_boxes_shape`'s padding entries; that PR is documentation and tests only, so this is out of scope there. Same family as #4175 (`merge` keeps stale `_N`): one is stale metadata, this one is dropped metadata, and one PR that decides how `_N` propagates can close both. Part of the #4020 audit.

## Environment

kornia 0.9.0rc1 @ `e0969438` (main), torch 2.9.1, python 3.11, cpu.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
